firedrake/mesh.py

```diff
@@ -106,6 +106,15 @@ def subset(self, markers):
         try:
             return self._subsets[markers]
         except KeyError:
+            # check that the given markers are valid
+            for marker in markers:
+                if marker not in self.markers:
```

Comment:
OK, the problem here is that the Facets instantiated on extruded meshes didn't supply the unique_markers argument /at all/.

What happens if you include this patch as well?

```diff
@@ -767,7 +775,8 @@ class ExtrudedMesh(Mesh):
                                         "exterior",
                                         exterior_f.facet_cell,
                                         exterior_f.local_facet_number,
-                                        exterior_f.markers)
+                                        exterior_f.markers,
+                                        unique_markers=exterior_f.unique_markers)
 
         self.ufl_cell_element = ufl.FiniteElement("Lagrange",
                                                   domain=mesh.ufl_cell(),
```